kademlia/src/test/java/ingsw/group1/kademlia/NodesRoutingTableTest.java

```diff
@@ -0,0 +1,265 @@
+package ingsw.group1.kademlia;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.BitSet;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link NodesRoutingTable} using generic {@link Node<BinarySet>} as Mock object
+ *
+ * @author Giorgia Bortoletti
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class NodesRoutingTableTest {
+
+    public static final int NUMBER_BITS_KEY = 3;
+
+    private NodesRoutingTable rt;
+    private PeerNode nodeOwner;
+    private Node<BinarySet> node;
+    private BitSet bitSet;
+    private BinarySet distance;
+
+    @Before
+    public void createRoutingTable() {
+        bitSet = BitSet.valueOf(new byte[]{(new Integer(7)).byteValue()});
+        nodeOwner = mock(PeerNode.class);
+        when(nodeOwner.getKey()).thenReturn(new BinarySet(bitSet));
+        node = mock(Node.class);
+        rt = new NodesRoutingTable(nodeOwner, NUMBER_BITS_KEY);
+    }
+
+    /**
+     * Calculating distance of node respect of nodeOwner using getDistance of PeerNode
+     *
+     * @param nodeOwner {@link Node<BinarySet>}
+     * @param node {@link Node<BinarySet>}
+     * @return BinarySet represents distance between two parameters
+     */
+    private BinarySet getDistancePeerNode(Node<BinarySet> nodeOwner, Node<BinarySet> node) {
+        return new PeerNode(nodeOwner.getKey()).getDistance(node);
+    }
+
+    /**
+     * Calculating distance of node respect of nodeOwner using getDistance of ResourceNode
+     *
+     * @param nodeOwner {@link Node<BinarySet>}
+     * @param node {@link Node<BinarySet>}
+     * @return BinarySet represents distance between two parameters
+     */
+    private BinarySet getDistanceResourceNode(Node<BinarySet> nodeOwner, Node<BinarySet> node) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unused private methods such as 'getDistanceResourceNode(Node,Node)'.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-kademlia/pullRequest?prid=4745729)

Reply:
This is written for future implementation.